cmd/controller-manager/app/controllermanager.go

```diff
@@ -183,23 +183,25 @@ func startClusterController(ctx controllerscontext.Context) (enabled bool, err e
 	mgr := ctx.Mgr
 	opts := ctx.Opts
 
+	if err := cluster.IndexField(mgr); err != nil {
```

Comment:
Seems we still don't need to move this out, since the only place uses the index is also controlled by `EnableTaintManager`:
```
// check if target cluster is removed from all bindings.
	if c.EnableTaintManager {
```

Reply:
I'm afraid not, we need this indexer in `isTargetClusterRemoved`.